### PR TITLE
Add smartseq2 query for v5 metadata

### PR DESCRIPTION
**Pull request by samanehsan**
Add query for smartseq2 bundles that follows the v5 metadata schema: https://github.com/HumanCellAtlas/metadata-schema/tree/develop

### CODE REVIEW

**Comment by hannes-ucsc** [comment]
This looks like it's not the final version we came up with yesterday?

File: scripts/v5_queries/smartseq2-query.json [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+            "files.process_json.biomaterial_collection.dissociation_process.dissociation_method": "FACS"
+          }
+        },
+        {
+          "match": {
+            "files.process_json.biomaterial_collection.dissociation_process.dissociation_method": "mouth-pipetting"
+          }
+        }
+      ],
+      "must_not": [
```
Comment (malloryfreeberg) [inline scripts/v5_queries/smartseq2-query.json]: I don't think this `must_not` will work. But, I need to add an analysis block to the current v5 process bundle so we can test this.

**Review by hannes-ucsc — APPROVED** [review]

**Comment by hannes-ucsc** [comment]
Thanks to @mikebaumann for his clever (and impressively clairvoyant) suggestion of using `must-not` and `range`.

File: scripts/v5_queries/smartseq2-query.json [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+          }
+        },
+        {
+          "match": {
+            "files.process_json.processes.content.paired_ends": true
+          }
+        },
+        {
+          "match": {
+            "files.biomaterial_json.biomaterials.content.biomaterial_core.ncbi_taxon_id": 9606
```
Comment (rexwangcc) [inline scripts/v5_queries/smartseq2-query.json:18]: I think it is good for us to keep both the `must` and the `must not` for readability.

Comment (rexwangcc) [reply scripts/v5_queries/smartseq2-query.json:18]: Even they are functionally equal to each other.

File: scripts/v5_queries/smartseq2-query.json [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        {
+          "match": {
+            "files.process_json.biomaterial_collection.dissociation_process.dissociation_method": "mouth pipette"
+          }
+        }
+      ],
+      "must_not": [
+        {
+          "exists": {
+            "field": "files.analysis_process_json"
```
Comment (rexwangcc) [inline scripts/v5_queries/smartseq2-query.json]: Are there any further changes required for this field?

File: scripts/v5_queries/smartseq2-query.json [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+            "files.process_json.biomaterial_collection.dissociation_process.dissociation_method": "FACS"
+          }
+        },
+        {
+          "match": {
+            "files.process_json.biomaterial_collection.dissociation_process.dissociation_method": "mouth pipette"
+          }
+        }
+      ],
+      "must_not": [
```
Comment (malloryfreeberg) [inline scripts/v5_queries/smartseq2-query.json]: I think this must_not exists field needs to be 
`"field": "files.process_json.analysis.analysis_process"`

**Review by rexwangcc — APPROVED** [review]
LGTM. Special thanks to @mikebaumann who already provided the perfect solution to filter out human id 4 months ago!

**Comment by malloryfreeberg** [comment]
I think the analysis must_not exists field needs to be:
"field": "files.process_json.analysis.analysis_process"

**Comment by samanehsan** [comment]
Just fixed that field @malloryfreeberg! Thanks!

**Comment by malloryfreeberg** [comment]
Great! There is now an example v5 bundle with an analysis process section here: https://github.com/HumanCellAtlas/metadata-schema/tree/malloryfreeberg-post_analysis_v5_bundle_example/examples/bundles/v5/Q4DemoSS2PostAnalysis

Once it's put in the data store, you can test the query out!

**Review by malloryfreeberg — APPROVED** [review]
Looks great!

File: scripts/v5_queries/smartseq2-query.json (hunk 4 above)
Comment (hannes-ucsc) [reply scripts/v5_queries/smartseq2-query.json]: This does work:

```
          {
            "match": {
              "files.process_json.processes.content.process_type.text": "analysis"
            }
          },
```

File: scripts/v5_queries/smartseq2-query.json [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        {
+          "match": {
+            "files.biomaterial_json.biomaterials.content.biomaterial_core.ncbi_taxon_id": 9606
+          }
+        }
+      ],
+      "should": [
+        {
+          "match": {
+            "files.process_json.biomaterial_collection.dissociation_process.dissociation_method": "FACS"
```
Comment (hannes-ucsc) [inline scripts/v5_queries/smartseq2-query.json]: This doesn't look right. It's missing `.processes.content` after `process_json`

File: scripts/v5_queries/smartseq2-query.json [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+          }
+        },
+        {
+          "match": {
+            "files.process_json.biomaterial_collection.dissociation_process.dissociation_method": "mouth pipette"
```
Comment (hannes-ucsc) [inline scripts/v5_queries/smartseq2-query.json]: ditto

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]

File: scripts/v5_queries/smartseq2-query.json (hunk 5 above)
Comment (malloryfreeberg) [reply scripts/v5_queries/smartseq2-query.json]: Yes, I think this needs to be:
`files.process_json.processes.content.enrichment_method`

Comment (samanehsan) [reply scripts/v5_queries/smartseq2-query.json]: Can you explain why the field should be `enrichment_method` @malloryfreeberg? This was originally `files.assay_json.content.single_cell.cell_handling` in the v4 query and that maps to `files.process_json.processes.content.biomaterial_collection.dissociation_process.dissociation_method` according to the field mapping spreadsheet

**Comment by samanehsan** [comment]
@hannes-ucsc I made the changes you requested! This query now matches the example v5 bundle in the data storage service (and not the one with the analysis process section)

**Review by hannes-ucsc — APPROVED** [review]

**Pull request merged**